src/Connector.SqlServer/Connector/IBulkSqlClient.cs

```diff
@@ -1,11 +1,12 @@
-﻿using System.Data;
-using System.Threading.Tasks;
+﻿using CluedIn.Connector.SqlServer.Utility;
 using CluedIn.Core.Connectors;
+using System.Data;
+using System.Threading.Tasks;
 
 namespace CluedIn.Connector.SqlServer.Connector
 {
     public interface IBulkSqlClient : ISqlClient
     {
-        Task ExecuteBulkAsync(IConnectorConnection config, DataTable table, string containerName);
+        Task ExecuteBulkAsync(IConnectorConnection config, DataTable table, SanitizedSqlName tableName);
```

Comment:
> Sanitize the string in the calling function and pass the santized string. eg: `var tableName = containerName.Sanitize();`
> 
> Add a guard clause to ExecuteBulkAsync() eg: `if(!containerName.IsSanitized) throw SqlException("{containerName} has an invalid name.")`

Besides that adding extensions to core types is usually bad practice, first looks for me just as syntax sugar.
But the second is really interesting, how it could work? This one `containerName.IsSanitized()`